PWGDQ/Tasks/dqEfficiency.cxx

```diff
@@ -726,6 +735,114 @@ struct AnalysisSameEventPairing {
     } // end loop over barrel track pairs
   }   // end runPairing
 
+  // Template function to run pair - lepton combinations
+  template <uint32_t TEventFillMap, uint32_t TTrackFillMap, typename TEvent, typename TTracks, typename TEventsMC, typename TTracksMC>
+  void runDileptonLepton(TEvent const& event, TTracks const& tracks, TEventsMC const& eventsMC, TTracksMC const& tracksMC)
```

Comment:
Having this function in the AnalysisSameEventPairing is breaking a bit the analysis model we used in the data analysis task (tableReader). It is also duplicating code since you are repeating the loop to build dilepton candidates which are produced in runPairing. So since we recently had an update in runPairing to produce a table which contains the indices of the lepton legs, its natural to make a separate task that subscribes to this table. You would then just need to loop over the dileptons in that table and over the selected tracks for your third leg. I think this is logically more simple, and becomes identical to the DileptonHadron task we have in tableReader.
We need also to generalize this part, since there are already ongoing barrel analyses that use this analysis topology.